src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -58,11 +62,74 @@ public class KNNQueryBuilder extends AbstractQueryBuilder<KNNQueryBuilder> {
     private final String fieldName;
     private final float[] vector;
     private int k = 0;
+    private Float distance = null;
     private QueryBuilder filter;
     private boolean ignoreUnmapped = false;
 
     /**
-     * Constructs a new knn query
+    * Constructs a new query with the given field name and vector
+     *
+     * @param fieldName Name of the field
+     * @param vector    Array of floating points
+    */
+    public KNNQueryBuilder(String fieldName, float[] vector) {
+        if (Strings.isNullOrEmpty(fieldName)) {
+            throw new IllegalArgumentException("[" + NAME + "] requires fieldName");
+        }
+        if (vector == null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires query vector");
+        }
+        if (vector.length == 0) {
+            throw new IllegalArgumentException("[" + NAME + "] query vector is empty");
+        }
+        this.fieldName = fieldName;
+        this.vector = vector;
+    }
+
+    /**
+     * Builder method for k
+     *
+     * @param k K nearest neighbours for the given vector
+     */
+    public KNNQueryBuilder k(int k) {
+        if (k <= 0 || k > K_MAX) {
+            throw new IllegalArgumentException("[" + NAME + "] requires 0 < k <= " + K_MAX);
+        }
+        if (distance != null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires either k or distance must be set");
+        }
+        this.k = k;
+        return this;
+    }
+
+    /**
+     * Builder method for distance
+     *
+     * @param distance the distance threshold for the nearest neighbours
+     */
+    public KNNQueryBuilder distance(Float distance) {
+        if (distance == null || distance < 0) {
+            throw new IllegalArgumentException("[" + NAME + "] requires distance >= 0");
+        }
+        if (k != 0) {
```

Comment:
@jmazanec15 Do you see any potential backward incompatibility if we change k from int to integer?

Reply:
We're still using int for k in KNNQueryBuilder